hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/PortableContextImpl.java

```diff
@@ -74,15 +74,16 @@ public ClassDefinition lookupClassDefinition(Data data) throws IOException {
         if (!data.isPortable()) {
             throw new IllegalArgumentException("Data is not Portable!");
         }
-
-        BufferObjectDataInput in = serializationService.createObjectDataInput(data);
-        int factoryId = in.readInt();
-        int classId = in.readInt();
-        int version = in.readInt();
-
-        ClassDefinition classDefinition = lookupClassDefinition(factoryId, classId, version);
-        if (classDefinition == null) {
-            classDefinition = readClassDefinition(in, factoryId, classId, version);
+        ClassDefinition classDefinition = null;
```

Comment:
What do you think about returning `classDefinition` in try-with-resources? This way you could limit the scope of the `classDefinition` variable

Reply:
Thanks for the suggestion, I've made the change. Please take a look.